client/src/pages/Programmes/ExistingProgramme.js

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useState } from "react";
+import { Flex, Text, Box, IconButton } from "@chakra-ui/core";
+import { OutlineButton } from "../../components";
+import MomentumClient from "../../common/MomentumClient";
+import { useHistory } from "react-router-dom";
+import ConfirmationDialog from "../../components/ConfirmationDialog";
+
+const ExistingProgramme = ({ id, notifyUpdate }) => {
+    const history = useHistory();
+
+    const [programmeDegreeInfo, setProgrammeDegreeInfo] = useState({});
+    const [openConfirmationDialog, setOpenConfirmationDialog] = useState(false);
+
+    useEffect(() => {
+        let isMounted = true;
+
+        Promise.all([MomentumClient.getProgrammeInfo(id)]).then((values) => {
+            if (!isMounted) {
+                return;
+            }
+
+            const [programmeDegreeInfo] = values;
+            setProgrammeDegreeInfo(programmeDegreeInfo);
+        });
+
+        return () => {
+            isMounted = false;
+        };
+    }, [id]);
+
+    const deleteProgramme = () => {
+        setOpenConfirmationDialog(false);
+
+        MomentumClient.deleteProgramme(programmeDegreeInfo._id);
+        history.push("/programmes/");
+        notifyUpdate();
```

Comment:
Since `MomentumClient.deleteProgramme()` is async, do you think putting the code once it has finished is a better idea?